README.md

```diff
@@ -1,20 +1,43 @@
 # \*IPs - Star Improvement Proposals
 *by Zachary Belford, Shane Jonas, Donald McIntyre, Yaz Khoury, Anthony Lusardi, Tomaz Kariz*
 
-Star Improvement Proposals are an open standard for contributors to propose changes, additions, or deprecations of core protocol specifications.
+Star Improvement Proposals is an extension of the BIP/EIP/ECIP process, with some changes to allow for decentralization of the process. Its fundamental change involves determinism of proposal numbers and content-based versioning. 
 
 The Star is meant to be replaced with the prefix/accronym of the users choice, though the prefix MUST be the same for all participants of the particular protocol specification.
 
 For example, All forks of the Star Improvement Proposal relating to the bitcoin network MUST be called `BIP`. The differentiator being the location/domain/github organization to which the \*IP belongs.
 
-Star Improvement Proposals are versioned by the content of their name. That is, in the case of making a proposal, the `version` identifier of the proposal MUST be derived by the last 8 characters of the SHA-3-512 of the proposal name. This `version` naming scheme is deterministic in that groups or individuals may choose to include another's proposal without any numbering conflicts.
+# Changes to BIP/EIP/ECIP
+## Version
+Star Improvement Proposals are versioned by the content of their name. First we create a hash of the preamble, then we use a substring of it to create the version hash.
 
-Once you have your \*ip repo up, people submit proposals to it via pull requests, adding a file named {networkName}IP-{proposalName}-{version}. Once the proposal is implemented by your team / group, you may signal your readyness by merging the pull request.
+### Preable hash
+SHA-3-512 of the proposal [preamble](https://github.com/ethereumclassic/ECIPs/blob/master/ECIP-1.sample.md#what-is-an-ecip).
+
+### Version Hash
+MUST be derived by the last 16 characters of the [Preamble Hash](#preamble-hash). This `version` naming scheme is deterministic in that groups or individuals may choose to include another's proposal without any numbering conflicts.
+
+## Proposal Number
+Star improvement proposals will retain a human friendly proposal number. However, this number will be derived from the hash such that it remains deterministic. The proposal number will be derived as the trailing 5 integers from the [Preamble Hash](#preamble-hash).
```

Comment:
What happens in case less than 5 integers occur in the hash?

Reply:
The preamble hash will be a raw 256 digit number, from there the last 16 digits will be the version number.

Separately, from the same original preamble hash raw 256 digit number above, there will be 5 integers used starting from the end. 

Although possible, we think it is highly unlikely that a full 256 digit hash will be all letters or have more than 251 letters and no integers.

Reply:
Because it is a possibility maybe we can have the teams, when that happens, just change something of the title, like a comma or any little thing in the preamble so that they get some integers.

Reply:
What he is saying is, the odds of getting less than 5 numbers in a sha-3-512 is very low.

Reply:
happy to hear any other ideas for having a numerical system to compliment. Personally, Im happy to leaave numbers out all together and use hashes.

Reply:
Is the resulting has a hex number? If so, what if to generate the 5 digit "human friendly" number it is turned into decimal and the last 5 digits of that one are used?

Reply:
Example of the above:

This is a sha3-512 hash:

1d76395115073f764ba4f206728d2203716809559196a59d9deddacbe646d9e5d6ce8c9868f7d837aa1bb20ce835426d10bc1c71319e44908e602ef7cd493d96

This is the same, but in decimal:

1543040303469948843454720447006790619894030419161739408629814013195114781995691563889852368227044919401394623845086971770484212387093490397553614790409622

So we can use:

ECIP-09622

Again, I think there is a 1 in 99999 probability that the 5 digit decimal number will collide with another ECIP. But I think that is not that bad. In that case the 16  digit has can be used when/if that happens.

Reply:
Make it [BASE58](https://en.wikipedia.org/wiki/Base58) so you get more density while still being human readable. Bonus is that it's a Bitcoin-like address scheme.

Reply:
Could you make a PR into this branch @pyskell ?

Reply:
Done @BelfordZ

Reply:
I have a proposal for what @DonaldMcIntyre pointed out about collision. In the unlikely chance we experience a collision with a pre-existing human-readable number, I suggest we extract a 6th trailing integer. 

This solves 2 problems: 
1) Having to place the collided number in a +1 nearest-neighbor slot if it's empty, a la a hash table. (Very hard to derive then from hash because it'll lead to wrong Human-Readable Number). 
2) It solves the long-term event of having occupied nearly all numbers in 5-digit starIP slots and wanting to expand further. Already opening up to a 6th digit due to collision can solve it.

Reply:
Resolving this as @pyskell's Base58 suggestion adds an extra density like he said and it has been merged into this branch.